.github/workflows/build-push.yml

```diff
@@ -67,18 +72,28 @@ jobs:
           aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
           aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
 
-      - name: Get the current release version
+      - name: Get next release version
         if: ${{ github.ref == 'refs/heads/master' }}
-        id: current-version
-        run: |
-          current_version=$(curl -s https://api.github.com/repos/${{ github.repository }}/releases/latest | jq -r .tag_name)
-          echo "Current version is $current_version"
-          echo "result=$current_version" >> $GITHUB_OUTPUT
+        uses: actions/github-script@v7
+        id: next-version
+        with:
+          result-encoding: string
+          script: |
+            const getNextVersion = require('./.branch-github-actions/custom-scripts/next-version');
```

Comment:
I'm not certain how this script works, will this check across all merged commits on main and take the highest semver?

Reply:
itll take the highest semver, then it will increment appropriately based on the commits it is releasing

commit tagged with major -> increment major version (2.86.0 -> 3.0.0)
commit tagged with minor -> increment minor version (2.86.0 -> 2.87.0)
no commits tagged with major or minor -> increment the right most (2.86.0 -> 2.86.1)